packages/pds/src/db/migrations/20230118T223059239Z-repo-sync-data.ts

```diff
@@ -2,24 +2,33 @@ import { chunkArray } from '@atproto/common'
 import { MemoryBlockstore } from '@atproto/repo'
 import { Kysely } from 'kysely'
 import { CID } from 'multiformats/cid'
-import DatabaseSchema from '../database-schema'
 import { RepoCommitBlock } from '../tables/repo-commit-block'
 import { RepoCommitHistory } from '../tables/repo-commit-history'
 
 const commitBlockTable = 'repo_commit_block'
 const commitHistoryTable = 'repo_commit_history'
 
-export async function up(db: DatabaseSchema): Promise<void> {
+export async function up(db: any): Promise<void> {
   await db.schema
     .createTable(commitBlockTable)
     .addColumn('commit', 'varchar', (col) => col.notNull())
     .addColumn('block', 'varchar', (col) => col.notNull())
-    .addPrimaryKeyConstraint(`${commitBlockTable}_pkey`, ['commit', 'block'])
+    .addColumn('creator', 'varchar', (col) => col.notNull())
+    .addPrimaryKeyConstraint(`${commitBlockTable}_pkey`, [
+      'commit',
+      'block',
+      'creator',
+    ])
     .execute()
   await db.schema
     .createTable(commitHistoryTable)
-    .addColumn('commit', 'varchar', (col) => col.primaryKey())
+    .addColumn('commit', 'varchar', (col) => col.notNull())
     .addColumn('prev', 'varchar')
+    .addColumn('creator', 'varchar', (col) => col.notNull())
+    .addPrimaryKeyConstraint(`${commitHistoryTable}_pkey`, [
+      'commit',
+      'creator',
+    ])
```

Comment:
Similar here— if we always expect to qualify these queries with a creator and want it to be cheap to enumerate commits/history by creator, then bumping `creator` up to the front of the index might be an easy win.

Reply:
good looks 👍 

did that in the sync-revamp PR as well: https://github.com/bluesky-social/atproto/pull/446/commits/4719922fea641a755c0235a715bade8d85f43edd